src/lib.ts

```diff
@@ -560,3 +550,29 @@ export function freeCrafts(): number {
     (have($skill`Expert Corner-Cutter`) ? 5 - get("_expertCornerCutterUsed") : 0)
   );
 }
+
+export function maxBy<T>(
```

Comment:
please add docstrings to this. we're using it everywhere and it's cool as shit but this is exactly the kind of thing that warrants a docstring, especially with your multiple extensions of the core functions and overrides to allow you to do weird shit.